File: 프로그래머스/lv1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.py
def solution(left, right):
    # 제곱수 
    result = 0

    # left부터 right까지 탐색
    for num in range(left, right + 1):
        isBool = True
        
        # 1일 때 예외 처리
        if num == 1:
            result -= num
            continue
        
        # num을 1부터 num까지 탐색하며 제곱수 검증, num은 range에서 제외
        for i in range(1, num):
            if i ** 2 == num:
                # 제곱수 있으면 result에서 빼줌
                result -= num
                isBool = False
        # 탐색 후 제곱수 없으면, result에 더해줌
        if isBool:
            result += num

    return result

File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.swift
// 약수의 개수가 홀수인 수 == 제곱수
// 제곱수는 빼고 나머지는 더한다
// left, right의 크기가 1000이라 O(n^2)해도 풀이가 가능할 듯

// 제곱수 판별 방법
// 1. n이 제곱수인지 판별하기 위해서 2 ~ n ** 1/2까지 제곱값과 비교
// 2. sqrt(n) == pow(Double(int(sqrt(n))), 2.0) 확인

import Foundation

func solution(_ left:Int, _ right:Int) -> Int {
    
    var sum = 0
    for num in left...right {
        var checker = false
        for numForCheck in 1...Int(pow(Double(num), 0.5)) {
            if num == Int(pow(Double(numForCheck), 2)) {
                sum -= num
                checker = true
            }
        }
        if !checker {
            sum += num
        }
    }
    
    return sum
}